src/swap.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { Problem } from "./cnd/axios_rfc7807_middleware";
+import { Cnd } from "./cnd/cnd";
+import { Swap, WalletError } from "./swap";
+
+describe("Swap", () => {
+  it("Throws a problem if cnd returns an error when executing an action", async () => {
+    const basePath = "http://example.com";
+    const cnd = new Cnd(basePath);
+
+    const selfPath = "/swap-id";
+
+    const swap = new Swap(cnd, selfPath, {});
+
+    cnd.fetch = jest.fn(async () => {
+      throw new Problem({ title: "" });
+    });
+
+    const promise = swap.tryExecuteSirenAction("action", {
+      maxTimeoutSecs: 1,
+      tryIntervalSecs: 0.01
+    });
+
+    await expect(promise).rejects.toBeInstanceOf(Problem);
+  });
+
+  it("Throws a WalletError if a wallet returns an error when resolving fields", async () => {
```

Comment:
You could just mock the `fieldValueResolver` instead of `Cnd`.

Reply:
The first step of `Swap.tryExecuteSirenAction` is `Swap.fetchDetails` so I do need to mock `Cnd` first. I am not sure to understand how you could test this without mocking `Cnd`.